packages/runtime/container-runtime/src/test/pendingStateManager.spec.ts

```diff
@@ -195,6 +205,26 @@ describe("Pending State Manager", () => {
 			assert(closeError === undefined);
 		});
 
+		it("empty batch is processed correctly", () => {
+			// Empty batch is reflected in the pending state manager as a single message
+			// with the following metadata:
+			submitBatch(
+				[
+					{
+						contents: JSON.stringify({ type: "groupedBatch", contents: [] }),
+						referenceSequenceNumber: 0,
+						metadata: { batchId: "batchId" },
+					},
+				],
+				1 /* clientSequenceNumber */,
+				{ emptyBatch: true },
+			);
+			// A groupedBatch is supposed to have nested messages inside its contents,
+			// but an empty batch has no nested messages. When processing en empty grouped batch,
+			// the psm will expect the next pending message to be an "empty" message as portrayed above.
+			process([], 1 /* batchStartCsn */, 3 /* sequenceNumber */);
```

Comment:
Bonus points for asserting that `savedOps` gets a new entry with sequenceNumber 3.  Bonus points are not really redeemable for anything though, so up to you ;)